### PR TITLE
Child workflow server side retry

### PR DESCRIPTION
**Pull request by mfateev**
Any child workflow that used RetryOptions after this change is going to use the server side retry instead of the client side one. The change is backwards compatible.

Also added an ability to replay workflows from a JSON file or resource to test backwards compatibility.

### CODE REVIEW

File: src/main/java/com/uber/cadence/internal/testservice/TestWorkflowMutableStateImpl.java [hunk 1]
```diff
@@ -525,7 +526,13 @@ private void processStartChildWorkflow(
   }
 
   // TODO
```
Comment (meiliang86) [inline src/main/java/com/uber/cadence/internal/testservice/TestWorkflowMutableStateImpl.java]: Remove this comment?

File: src/test/java/com/uber/cadence/workflow/WorkflowTest.java [hunk 2]
```diff
@@ -97,7 +98,7 @@
    * When set to true increases test, activity and workflow timeouts to large values to support
    * stepping through code in a debugger without timing out.
    */
-  private static final boolean DEBUGGER_TIMEOUTS = false;
+  private static final boolean DEBUGGER_TIMEOUTS = true;
```
Comment (meiliang86) [inline src/test/java/com/uber/cadence/workflow/WorkflowTest.java]: Revert this change?

File: src/main/java/com/uber/cadence/internal/worker/WorkflowWorker.java [hunk 3]
```diff
@@ -100,42 +101,58 @@ public void start() {
...
+    return queryWorkflowExecution(queryType, args, history, null);
+  }
+
+  private byte[] queryWorkflowExecution(
+      String queryType, byte[] args, WorkflowExecutionHistory history, byte[] nextPageToken)
+      throws Exception {
+    List<HistoryEvent> events = history.getEvents();
+    HistoryEvent startedEvent = events.get(0);
+    WorkflowExecutionStartedEventAttributes started =
+        startedEvent.getWorkflowExecutionStartedEventAttributes();
```
Comment (meiliang86) [inline src/main/java/com/uber/cadence/internal/worker/WorkflowWorker.java]: Move these three lines right before ```if (started == null) {``` (line 142)?

File: src/main/java/com/uber/cadence/internal/testservice/TestWorkflowMutableStateImpl.java (hunk 1 above)
Comment (mfateev) [reply src/main/java/com/uber/cadence/internal/testservice/TestWorkflowMutableStateImpl.java]: Implemented validation.
